nautobot/core/tests/test_authentication.py

```diff
@@ -383,7 +383,16 @@ def test_create_object(self):
         )
         obj_perm.users.add(self.user)
         obj_perm.object_types.add(ContentType.objects.get_for_model(Prefix))
-
+        related_obj_perm = ObjectPermission.objects.create(
+            name="Related object permission",
+            actions=["view", "add", "change"],
+        )
+        related_obj_perm.users.add(self.user)
+        related_obj_perm.object_types.add(
+            ContentType.objects.get_for_model(Namespace),
+            ContentType.objects.get_for_model(Status),
+            ContentType.objects.get_for_model(Location),
+        )
```

Comment:
Feels like we should also *add* some test steps here and below to verify the issue being fixed - i.e. user with `obj_perm` but without `related_obj_perm` should also be attempted and fail as well.

Reply:
It occurs to me also that we need to test the impact of this change specifically on GET and PATCH requests, for example:

- If I have view permissions for Location but not for Status, can I GET `/api/dcim/locations/` or does that now reject since I can't view the location `status`?
- If I have view and edit permission for Location but not for Status, can I PATCH `/api/dcim/locations/<UUID>/` with a new `name` value (not changing its `status`) or does that now reject?

Reply:
Added some tests to confirm these expected behaviors